db/mongo/client.go

```diff
@@ -148,6 +144,24 @@ func (c *client) ListUsers(limit int32, name string) ([]*domain.User, error) {
 	return userInfo, nil
 }
 
+func applyFilter(filterMap map[string]interface{}) map[string]interface{} {
+
+	for k, v := range filterMap {
+		switch mval := v.(type) {
+		case []string:
+			filterMap[k] = bson.M{"$in": mval}
+		case string:
+			// support searching by name using case-insensitive matching
+			fmt.Println("v value is = ", v)
+			if k == "name" {
+				filterMap[k] = bson.M{"$regex": primitive.Regex{Pattern: "^" + regexp.QuoteMeta(mval) + "$", Options: "i"}}
```

Comment:
what do you understand from this regex?

Reply:
It provides a generalized way to match patterns with sequences of characters. I used it in case of mongo data store, to get the desired results even in case if user search names by putting lower case letters & in db the name is available in upper case letters.